pkg/cmd/task/describe_test.go

```diff
@@ -238,6 +238,82 @@ Taskruns
 	test.AssertOutput(t, expected, out)
 }
 
+func TestTaskDescribe_NoStepName(t *testing.T) {
+	clock := clockwork.NewFakeClock()
+	cs, _ := test.SeedTestData(t, pipelinetest.Data{
+		Tasks: []*v1alpha1.Task{
+			tb.Task("task-1", "ns",
+				tb.TaskSpec(
+					tb.Step("", "mandriva"),
+				),
+				tb.TaskSpec(
+					tb.Step("alatouki", "stampede"),
+				),
+				tb.TaskSpec(
+					tb.Step("", "mandrake"),
+				),
+				tb.TaskSpec(
+					tb.Step("lamaracarena", "yggdrasil"),
+				),
+			),
+		},
+		TaskRuns: []*v1alpha1.TaskRun{
+			tb.TaskRun("tr-1", "ns",
+				tb.TaskRunStatus(
+					tb.StatusCondition(apis.Condition{
+						Status: corev1.ConditionFalse,
+						Reason: resources.ReasonFailed,
+					}),
+					tb.TaskRunStartTime(clock.Now()),
+					cb.TaskRunCompletionTime(clock.Now().Add(5*time.Minute)),
+				),
+			),
+		},
+		Namespaces: []*corev1.Namespace{
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "ns",
+				},
+			},
+		},
+	})
+
+	p := &test.Params{Tekton: cs.Pipeline, Clock: clock, Kube: cs.Kube}
+	task := Command(p)
+	clock.Advance(20 * time.Minute)
+	out, err := test.ExecuteCommand(task, "desc", "task-1")
+	if err != nil {
+		t.Errorf("Unexpected error: %v", err)
+	}
+	expected := `Name:        task-1
+Namespace:   ns
+
+Input Resources
+
+ No input resources
+
+Output Resources
+
+ No output resources
+
+Params
+
+ No params
+
+Steps
+
+ unnamed-0
+ alatouki
+ unnamed-1
```

Comment:
This would be `unnamed-2` on the *backend*, that's why I wanted to add this test :angel:

Reply:
Does it matter tho? It's generated name anyway?

Reply:
Well, it does if the name of the step we display here is no the same of the one the controller will generate when executing the `TaskRun`

Reply:
if only there was no auto generated step names.....